lab_3/main.py

```diff
@@ -110,42 +187,115 @@ def new_language(self, encoded_text: tuple, language_name: str) -> int:
         :param language_name: a language
         :return: 0 if succeeds, 1 if not
         """
-        pass
+        if not isinstance(encoded_text, tuple) or not isinstance(encoded_text[0], tuple) \
+                or not isinstance(language_name, str):
+            return 1
+        self.n_gram_storages[language_name] = {}
+        for level in self.trie_levels:
+            storage_language = NGramTrie(level)
+            storage_language.fill_n_grams(encoded_text)
+            storage_language.calculate_n_grams_frequencies()
+            storage_language.calculate_log_probabilities()
+            self.n_gram_storages[language_name][level] = storage_language
+        return 0
 
-    def _calculate_distance(self, first_n_grams: tuple, second_n_grams: tuple) -> int:
+    @staticmethod
+    def _calculate_distance(first_n_grams: tuple, second_n_grams: tuple) -> int:
         """
         Calculates distance between top_k n-grams
         :param first_n_grams: a tuple of the top_k n-grams
         :param second_n_grams: a tuple of the top_k n-grams
         :return: a distance
         """
-        pass
+        if not isinstance(first_n_grams, tuple) or not isinstance(second_n_grams, tuple) or\
+                len(first_n_grams) != 0 and not isinstance(first_n_grams[0], (str, tuple)):
+            return -1
+        if len(second_n_grams) != 0 and not isinstance(second_n_grams[0], (str, tuple)):
+            return -1
+
+        sum_distance = 0
+        list_all_first = []
+        list_all_second = []
+        try:
```

Comment:
can't we calculate distance in a more short way? I see some code duplications. If we can not - just try to remove such a big `try-except` usage.